Rakefile

```diff
@@ -131,28 +130,17 @@ task 'gem:native' do
         "invoked on macos with ruby #{RUBY_VERSION}. The ruby macos artifact " \
         "build should be running on ruby 2.5."
     end
-    system "rake cross native gem RUBY_CC_VERSION=#{ruby_cc_versions} V=#{verbose} GRPC_CONFIG=#{grpc_config}"
+    system "rake cross native gem RUBY_CC_VERSION=3.0.0:2.7.0:2.6.0:2.5.0:2.4.0:2.3.0 V=#{verbose} GRPC_CONFIG=#{grpc_config}"
   else
     Rake::Task['dlls'].execute
-    ['x86-mingw32', 'x64-mingw32'].each do |plat|
+    ['x86-mingw32', 'x64-mingw32', 'x86_64-linux', 'x86-linux'].each do |plat|
       run_rake_compiler plat, <<-EOT
         gem update --system --no-document && \
         bundle && \
         rake native:#{plat} pkg/#{spec.full_name}-#{plat}.gem pkg/#{spec.full_name}.gem \
-          RUBY_CC_VERSION=#{ruby_cc_versions} V=#{verbose} GRPC_CONFIG=#{grpc_config}
-      EOT
-    end
-    # Truncate grpc_c.*.ruby files because they're for Windows only.
-    File.truncate('grpc_c.32.ruby', 0)
-    File.truncate('grpc_c.64.ruby', 0)
-    ['x86_64-linux', 'x86-linux'].each do |plat|
-      run_rake_compiler plat,  <<-EOT
-        gem update --system --no-document && \
-        bundle && \
-        rake native:#{plat} pkg/#{spec.full_name}-#{plat}.gem pkg/#{spec.full_name}.gem \
-          RUBY_CC_VERSION=#{ruby_cc_versions} V=#{verbose} GRPC_CONFIG=#{grpc_config} &&
-        sudo chmod -R a+rw pkg &&
-        patchelf_gem.sh pkg/#{spec.full_name}-#{plat}.gem
```

Comment:
I guess patchelf_gem.sh part is not included in the mainstream so this might need to be here.

Reply:
I can't see what we need `patchelf_gem.sh` for now, though.

Here are the dynamic dependencies of the x86-64 pre-compiled linux ruby gems now:

```
~/grpc/pkg$ find grpc-1.35.0.dev-x86_64-linux -name grpc_c.so -exec bash -c "echo {} && ldd {}" \;
grpc-1.35.0.dev-x86_64-linux/src/ruby/lib/grpc/2.3/grpc_c.so
	linux-vdso.so.1 (0x00007ffe5ab59000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f236363b000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f2363636000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f23634b3000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f23632f2000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f2363dd8000)
grpc-1.35.0.dev-x86_64-linux/src/ruby/lib/grpc/2.7/grpc_c.so
	linux-vdso.so.1 (0x00007ffe4f14d000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f5f9ccc7000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f5f9cb06000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f5f9d5c6000)
grpc-1.35.0.dev-x86_64-linux/src/ruby/lib/grpc/2.6/grpc_c.so
	linux-vdso.so.1 (0x00007ffee61ee000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa3c049e000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa3c02dd000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa3c0d9d000)
grpc-1.35.0.dev-x86_64-linux/src/ruby/lib/grpc/2.5/grpc_c.so
	linux-vdso.so.1 (0x00007fffddeec000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f7e2967c000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f7e29677000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f7e294f4000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f7e29333000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f7e29e19000)
grpc-1.35.0.dev-x86_64-linux/src/ruby/lib/grpc/2.4/grpc_c.so
	linux-vdso.so.1 (0x00007ffc3a919000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fbd27850000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fbd2784b000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fbd276c8000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fbd27507000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fbd27fed000)
```

... note how there is no `libcrypt` dependency, which if I'm correct [is the purpose of patchelf_gem.sh](https://github.com/grpc/grpc/blob/393687c3b7faba823648e020b73fcfcba6a757f9/third_party/rake-compiler-dock/build/patchelf_gem.sh#L20) ?


Also note that [manylinux14 appears to have removed libcrypt from their list of permitted dynamic libraries](https://www.python.org/dev/peps/pep-0599/) -- "This list is identical to the externally-provided libraries originally allowed for manylinux2010, with one exception: libcrypt.so.1 was removed due to being deprecated in Fedora 30."

Reply:
Oh mnaylinux2014 somehow managed to remove that dependency. Actually it was only observed from ruby build case so this could be skipped if the grpc so files start not to have it. Thanks!

Reply:
I remove the unnecessary dependency to `libcrypt.so.2` [in rake-compiler-dock](https://github.com/rake-compiler/rake-compiler-dock/blob/9c6e35e91661a98335a0ecddd329f6c96a8a0a70/Dockerfile.mri.erb#L210-L211).